src/__tests__/stockSlice.js

```diff
@@ -0,0 +1,47 @@
+import stockReducer, { fetchStock } from '../redux/features/stock/stockSlice';
+
+const initialState = {
+  data: [],
+  keys: [],
+  Views: [],
+  laoding: false,
+  error: '',
+};
+
+it('Should have the correct initial state', () => {
+  expect(stockReducer(undefined, {})).toEqual(initialState);
+});
+
+it('Should handle the pending case correctly', () => {
+  expect(
+    stockReducer(initialState, fetchStock.pending()).laoding,
```

Comment:
* According to the requirements you are to Create unit tests for pure functions in your Redux code.
Create integration tests for your application using the React Testing Library.
You may need to mock the access to the API, so that your tests don't send actual requests.
You may need to mock the connection to the Redux Store. Kindly add more tests